core/src/main/scala/kafka/coordinator/group/MemberMetadata.scala

```diff
@@ -107,7 +107,7 @@ private[group] class MemberMetadata(var memberId: String,
    * Check if the provided protocol metadata matches the currently stored metadata.
    */
   def matches(protocols: List[(String, Array[Byte])]): Boolean = {
-    if (protocols.size != this.supportedProtocols.size)
+    if (protocols.lengthCompare(supportedProtocols) != 0)
```

Comment:
FYI reviewers -
Comparison of two iterables. Reduces comparison time complexity from O(this.size + that.size) to O(this.size min that.size)